plugins/catalog/src/components/CatalogTable/CatalogTable.tsx

```diff
@@ -75,40 +69,15 @@ export const CatalogTable = ({ columns }: CatalogTableProps) => {
     );
   }
 
-  const actions: TableProps<EntityRow>['actions'] = [
-    ({ entity }) => {
-      const url = getEntityMetadataViewUrl(entity);
-      return {
-        icon: () => <OpenInNew fontSize="small" />,
-        tooltip: 'View',
-        disabled: !url,
-        onClick: () => {
-          if (!url) return;
-          window.open(url, '_blank');
-        },
-      };
-    },
-    ({ entity }) => {
-      const url = getEntityMetadataEditUrl(entity);
-      return {
-        icon: () => <Edit fontSize="small" />,
-        tooltip: 'Edit',
-        disabled: !url,
-        onClick: () => {
-          if (!url) return;
-          window.open(url, '_blank');
-        },
-      };
-    },
-    ({ entity }) => {
-      const isStarred = isStarredEntity(entity);
-      return {
-        cellStyle: { paddingLeft: '1em' },
-        icon: () => favouriteEntityIcon(isStarred),
-        tooltip: favouriteEntityTooltip(isStarred),
-        onClick: () => toggleStarredEntity(entity),
-      };
-    },
+  const defaultActions: TableProps<EntityRow>['actions'] = [
+    ({ entity }) => actionFactories.createViewUrlAction(entity),
+    ({ entity }) => actionFactories.createEditUrlAction(entity),
+    ({ entity }) =>
+      actionFactories.createStarredAction(
+        entity,
+        isStarredEntity,
```

Comment:
Make sense. I was thinking to create an agnostic factory (something like `createAction`) to give a little bit of structure to the action, but then I realise that is pointless since we have typescript that force us to the interface `Action<EntityRow>`.

As far as I know, it's not possible to use context outside a React component, so I just restored the actions (`defaultActions`) in the component.

Let me know if it makes sense 👍